lib/src/tar/export.rs

```diff
@@ -320,6 +322,13 @@ impl<'a, W: std::io::Write> OstreeTarWriter<'a, W> {
                 &commitmeta,
             )?;
         }
+
+        // and add the canonical ref
+        let path: Utf8PathBuf = format!("{}/repo/refs/heads/ostree", OSTREEDIR).into();
```

Comment:
Then here drop the `/ostree`.

Reply:
(Also, this could be spelled `Utf8Path::new(OSTREEDIR).join("repo/refs/heads")`)

Reply:
> I think this should be `ostree/encapsulated` though; that's what the ref is. It's the string you'd pass to `ostree rev-parse`.
>
> Then here drop the `/ostree`.

That was my initial plan, but to have this code be generic over all values of `OSTREEREF`, I didn't think it was worth the added complexity of iterating over all the components under `heads/` of the ref and emitting tar entries for every subdir in the path. I *think* most extractors would correctly handle them missing, but it doesn't feel proper skipping them.

> (Also, this could be spelled `Utf8Path::new(OSTREEDIR).join("repo/refs/heads")`)

Hmm, the `format!(...).into()` pattern seems to be prevalent in this file. Should I change those too?

Reply:
> Hmm, the format!(...).into() pattern seems to be prevalent in this file. Should I change those too?

Nah it's fine, we can do that later.